src/realm-settings/RealmSettingsSection.tsx

```diff
@@ -164,22 +164,30 @@ export const RealmSettingsSection = () => {
   const [currentUser, setCurrentUser] = useState<UserRepresentation>();
   const { whoAmI } = useWhoAmI();
 
-  const kpComponentTypes =
-    useServerInfo().componentTypes!["org.keycloak.keys.KeyProvider"];
+  const kpComponentTypes = useServerInfo().componentTypes![KeyProviderType];
+
+  const sortByPriority = (components: ComponentRepresentation[]) => {
+    const sortedComponents = components?.sort(
```

Comment:
According to the function signature the `components` argument is always defined, so there should be no need to use optional chaining here.
```suggestion
    const sortedComponents = components.sort(
```

Reply:
Be aware that `.sort()` uses an [in-place](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) algorithm, this means that the original array will be changes as well. You might want to create a new array before operating on it.

For example:
```suggestion
    const sortedComponents = [...components].sort(
```

Or
```suggestion
    const sortedComponents = Array.from(components).sort(
```